otter/effect_dispatcher.py

```diff
@@ -28,3 +32,15 @@ def get_dispatcher(reactor):
         }),
         make_twisted_dispatcher(reactor),
     ])
+
+
+def get_full_dispatcher(reactor, authenticator, log, service_mapping, region):
```

Comment:
How is it different than having one dispatcher? `TenantScope` is not going to be used if it's effect is not going to be performed.